File: laramie-interface-webapi/src/domain/entities/permiso.ts
export default class Permiso {

    id: number;
    codigo: string;
    nombre: string;
    descripcion: string;
    sistema: string;
    idModulo: number;

    constructor(
        id: number = 0,
        codigo: string = "",
        nombre: string = "",
        descripcion: string = "",
        sistema: string = "",
        idModulo: number = 0)
    {
        this.id = id;
        this.codigo = codigo;
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.sistema = sistema;
        this.idModulo = idModulo;
    }

    setFromObject = (row) =>
    {
        this.id = row.id ?? 0;
        this.codigo = row.codigo ?? "";
        this.nombre = row.nombre ?? "";
        this.descripcion = row.descripcion ?? "";
        this.sistema = row.sistema ?? "";
        this.idModulo = row.idModulo ?? "";
    }

}

File: modules/seguridad/models/permiso.py
from sqlalchemy import Column, BigInteger, String
from sqlalchemy.orm import relationship

import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", ".."))
from shared.database import Base


class Permiso(Base):
    __tablename__ = "seguridad_permisos"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    codigo = Column(String(50), nullable=False, unique=True)
    nombre = Column(String(250), nullable=False)
    descripcion = Column(String(250), nullable=False, default="")
    sistema = Column(String(50), nullable=False, default="")
    id_modulo = Column(BigInteger, nullable=False, default=0)

    perfiles = relationship("Perfil", secondary="seguridad_perfil_permiso", back_populates="permisos", lazy="selectin")
